template_registry.go

```diff
@@ -65,18 +65,18 @@ func (r *TemplateRegistry) Dir(path ...string) (*TemplateRegistryDir, error) {
 	if len(path) == 1 {
 		return dir, nil
 	}
-	dir, err := dir.Dir(path[1:]...)
+	dir, err := dir.Mkdir(path[1:])
```

Comment:
Yeah, definitely. I propose to ignore the existence of the directory in Mkdir functions, so that if a directory does not exist, we return the directory that has just been created, or, if a directory already existed, we return that directory.

Reply:
ok, the only case when we need to return an error is when a dir path is also a path to an existing template, so there is a naming conflict between the sub dir and the template